### ISSUE TITLE
Improve feedback in "Starting" phase

### ISSUE THREAD

**Issue by joepavitt** [flowforge/flowforge#2487]
### Description

Watching back a lot of the users that are dropping off having created an instance, but then never opening the Editor, pretty much all of them hover around the "Starting" pill, get bored, move away, and never come back.

It's not like it's taking long either though, in most cases, they're giving up after 20 seconds or so.

A suspicion I have is that the "Starting" pill does look a little static, and doesn't give strong indication that anything is going on. My thoughts are two fold:

- Have a small animation sequence inside the "Starting" pill to show that it's not frozen
- Send them to the Node-RED Logs for that Instance to show activity, or the Instance Overview (where the Logs are also present)
    - As an alternative here, send them to the Node-RED Instance Overview, where they will see the activity and an option for the "logs"

### Have you provided an initial effort estimate for this issue?

I have provided an initial effort estimate

**Comment by joepavitt** [comment]
fyi @MarianRaphael - think this could help our onboarding conversion, and (hopefully) overall stickiness for churn

**Comment by iskerrett** [comment]
Fwiw, the first couple of times I used FlowForge Cloud I completely missed the 'starting' status. I was wondering what was going on?  I think making this much more obvious would definitely help.

**Comment by joepavitt** [comment]
Thanks or flagging @iskerrett - really useful feedback.

**Comment by joepavitt** [comment]
Have tagged this as high priority as this is a big cliff of users dropping off and should be looked at sooner rather than later (fyi @MarianRaphael)

**Comment by joepavitt** [comment]
Separating "send them to the Node-RED Instance Overview as it stands, where they will see the activity and an option for the "logs" into a standalone task as that's a quick win we can put in now.

**Comment by MarianRaphael** [comment]
Additional ideas:
-  A countdown next or inside the "Starting" pill (for the first auto-created instance)
-  A notification pop-up when the instance is ready

**Comment by hardillb** [comment]
No

There is a lot of things going on before we can get anywhere near showing logs.

Assuming this is FF cloud (or any other kuberneate instance)

The first time a stack is used on a give node it needs to be pulled from the container registery, this can take minuets, there is absolutely nothing to show during this time. The pill shows "starting" iirc

Once pulled it will start the launcher, which will then connect to the forge app to get the instance details. At this point there is the first entry in the logs ("Launcher started", "Loading project settings")

Next is goes into installing mode, where it starts to install the nodes the instance needs. (again this can take a long time) but the pill should say installing now.

Once that completes then it will start Node-RED

**Comment by joepavitt** [comment]
Okay, so the two UI states are:

- starting: represents 2 stages:
    - pulling container (no logs or updates)
    - starting the launcher (logs become available)
- installing: dependent upon individual node setup, but we do know this state well and have logs

**Comment by joepavitt** [comment]
Steve proposed faking logs when we are in starting phase 1a

**Comment by Pezmc** [comment]
Other ideas mentioned at the engineering meeting (for outside of this release):

 - Using service workers to send a browser push notification when ready
 - Having some sort of tutorial video or demo version of NR for users to spend time on while the instance starts
 - Tracking instance creation and sending an email if the user doesn't access

**Comment by joepavitt** [comment]
> Tracking instance creation and sending an email if the user doesn't access

fyi @robmarcer - we're losing 50% of users that create an Instance atm, but never open it. Could be a good campaign for CS.

**Comment by joepavitt** [comment]
> Having some sort of tutorial video or demo version of NR for users to spend time on while the instance starts

For this, @robmarcer or @Hisma it'd be great to have a "Getting started with Node-RED" video we can point them to, any thoughts? We know 34% of users are there for educational purposes, so is a useful resource for them.

**Comment by robmarcer** [comment]
@joepavitt thanks, that number is really high so I agree it's worth working on. I can't currently tell from the data in FF Cloud which users have opened the editor but I can tell who has deployed a flow and how many times. I'll create a new CS email which is sent to those who have not deployed a flow, giving them some pointers on what to try next.

**Comment by joepavitt** [comment]
> I'll create a new CS email which is sent to those who have not deployed a flow, giving them some pointers on what to try next.

Perfect for now. Thanks

**Comment by joepavitt** [comment]
> for example adding a video as suggested.

@robmarcer was actually thinking of linking to video directly from FF. We need to keep users busy for the couple of minutes it's taking to spin up the first instance of Node-RED.

**Comment by iskerrett** [comment]
Why can't we just have a graphic brought up that says 'Welcome to FlowForge. Give us a couple of minutes while we spin up your instance of Node-RED.'  Have a status bar moving (I know it won't be accurate) that tells people work is still being done.

One the instance is ready, change the graphic to 'Your Node-RED instance is ready. Open the editor here".  Include a link to the editor.

**Comment by joepavitt** [comment]
> Why can't we just have a graphic brought up that says 'Welcome to FlowForge. Give us a couple of minutes while we spin up your instance of Node-RED.' Have a status bar moving (I know it won't be accurate) that tells people work is still being done.

I did ask @Pezmc for this on Tuesday but it was deemed to close to Release to fit it in, for a couple of reasons:

1. We have no way of knowing whether it's the first time someone has logged in. This would have required a new DB table in order to start tracking this. Which I propose we should still be doing, just not in time for today.
2. For the progress bar, we have 0 indiciation as to how long the process will take, nor any feedback on how far along that process is. I am in favour of giving some more visual clue that _something_ is happening though, which is what I proposed [here](https://github.com/flowforge/flowforge/issues/2487#issuecomment-1659828112).

**Comment by iskerrett** [comment]
Sorry I missed the animation block. FWIW, when I just created my account I was not on this page. I was taken to the application page.  Not sure if that matters or not?

**Comment by Pezmc** [comment]
Yes, indeed, instances are only created after a user _verifies_ their email address (to make it slightly harder to spin up hundreds of instances maliciously).

**Comment by joepavitt** [comment]
> Not sure if that matters or not

I did ask about us being able to redirect straight to the new Insgance overview, but given we had no way of setting context to the user of "welcome to your new instance", then it felt a little odd that they wouldn't land on the standard landing page they'd see when logging in.
